### ISSUE TITLE
Add previous pregnancy record to Obsetric History on the Progress Report [2h]

### ISSUE THREAD

**Issue by adamhstewart** [TIP-Global-Health/eheza-app#1326]
## Issue 
Page 1 of Obstetrical History asks about term pregnancies, preterm deliveries, preterm and term stillbirths, and abortions. For any response >0, the progress report should state under Obstetric History (renamed from "Risk Factors:
- Number of abortions (<20 weeks)
- Number of preterm stillbirths
- Number of term stillbirths
- Number of preterm deliveries

<img width="1004" alt="Cursor_and_E-Heza_System" src="https://github.com/user-attachments/assets/5a455d57-a06f-4ac4-af39-9be0b88c5274">

## Tasks
- Rename the section
- add the information

**Comment by anvmn** [comment]
@LudmilaGoichman please test on Nutrition env

**Comment by LudmilaGoichman** [comment]
@anvmn 

- Nutrition env
- Android tablet

### Questions/Found Issues:
1. Should these answers be included in the report?

<img width="142" alt="image" src="https://github.com/user-attachments/assets/f3dc691f-a2f9-4bf1-97f4-7137aa7197f7">| <img width="354" alt="image" src="https://github.com/user-attachments/assets/bc83590e-7988-4270-bf73-4e30be4b9e27">
---|---

2. When the first page changes to 0 and the second page is `RED`, the answers from the second page will appear in the report.

<img width="142" alt="image" src="https://github.com/user-attachments/assets/f3dc691f-a2f9-4bf1-97f4-7137aa7197f7">| <img width="354" alt="image" src="https://github.com/user-attachments/assets/28203312-fa95-4559-b40f-23777960ab51">|<img width="354" alt="image" src="https://github.com/user-attachments/assets/ba5d1d47-104e-4e85-8639-9021ddbf4658">
---|---|---

### Checked and Approved:
1. The following options appear in the report:

> Number of abortions (<20 weeks)
Number of preterm stillbirths
Number of term stillbirths
Number of preterm deliveries 

<img width="278" alt="image" src="https://github.com/user-attachments/assets/e2e3e014-73c7-4059-9305-c62c304a2aeb">|<img width="351" alt="image" src="https://github.com/user-attachments/assets/38c1bc5b-4168-4bef-9088-e27332ea8b46">
---|---

2. When Gravida 01 and Para 0000, the report is empty

<img width="315" alt="image" src="https://github.com/user-attachments/assets/4afcfae3-bcea-40a6-99d3-05f3b0661e4d">|<img width="351" alt="image" src="https://github.com/user-attachments/assets/76dbc211-eb7d-4fa7-a157-6090f28d0b56">
---|---

**Comment by anvmn** [comment]
@LudmilaGoichman 

> Should these answers be included in the report?

yes, we're adding new entries - all that was there, remains.

> When the first page changes to 0 and the second page is RED, the answers from the second page will appear in the report.

It's an edge case. But I suppose that if answers were given, we should present them at the report.

**Comment by LudmilaGoichman** [comment]
@anvmn 

>> When the first page changes to 0 and the second page is RED, the answers from the second page will appear in the report.

> It's an edge case. But I suppose that if answers were given, we should present them at the report.

But when Gravida = 1 Para = 0000, the second page doesn't appear in Obstetric History.

**Comment by adamhstewart** [comment]
> It's an edge case. But I suppose that if answers were given, we should present them at the report.

@anvmn is this result possible because the answers on the second page were filled out prior to the questions on the first page resulting in Para of 0000? If so, we can move this over to acceptance.

**Comment by anvmn** [comment]
@adamhstewart 

> > It's an edge case. But I suppose that if answers were given, we should present them at the report.
> 
> @anvmn is this result possible because the answers on the second page were filled out prior to the questions on the first page resulting in Para of 0000? If so, we can move this over to acceptance.

Correct.  What can happen is that Para is not 0000, and then second step is presented.
Second step is save and then you go back and change Para to 0000.
Edge case, but I think that it would be correct to should show the entries.

**Comment by adamhstewart** [comment]
@anvmn agreed ^^
